configuration.rst

```diff
@@ -186,111 +215,353 @@ This parameter is then referenced in the framework config in
+#. ``config/packages/services.yaml`` (and ``services.xml`` and ``services.php`` files too);
+
+Take the ``framework`` package, installed by default, as an example:
+
+* First, ``config/packages/framework.yaml`` is loaded in all environments and
+  it configures the framework with some options;
+* In the **prod** environment, nothing extra will be set as there is no
+  ``config/packages/prod/framework.yaml`` file;
+* In the **dev** environment, there is no file either (
+  ``config/packages/dev/framework.yaml`` does not exist).
+* In the **test** environment, the ``config/packages/test/framework.yaml`` file
+  is loaded to override some of the settings previously configured in
+  ``config/packages/framework.yaml``.
+
+.. seealso::
+
+    See the ``configureContainer()`` method of
+    :doc:`the Kernel class </configuration/front_controllers_and_kernel>` to
+    learn everything about the loading order of configuration files.
+
+.. _selecting-the-active-environment:
+
+Selecting the Active Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony applications come with a file called ``.env`` located at the project
+root directory. This file is used to define the value of environment variables
+and it's explained in detail :ref:`later in this article <config-dot-env>`.
+
+Open the ``.env`` file and edit the value of the ``APP_ENV`` variable to change
+the environment in which the application runs. For example, to run the
+application in production:
+
+```bash
+# .env
+APP_ENV=prod
+```
+
+This value is used both for the web and for the console commands. However, you
+can override it for commands by setting the ``APP_ENV`` value before running them:
+
+.. code-block:: terminal
+
+    # Use the environment defined in the .env file
+    $ php bin/console command_name
+
+    # Ignore the .env file and run this command in production
+    $ APP_ENV=prod php bin/console command_name
+
+.. deprecated:: 4.2
+
+    In previous Symfony versions you could configure the environment with the
+    ``--env`` command option, which was deprecated in Symfony 4.2.
+
+Creating a New Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default three environments provided by Symfony are enough for most projects,
+but you can define more environments if you want to. For example, this is how
+you can define a ``staging`` environment where the client can test the project
+before going to production:
 
-For more information about parameters - including how to reference them from inside
-a controller - see :ref:`service-container-parameters`.
+#. Create a configuration directory with the same name as the environment (in
+   this case, ``config/packages/staging/``);
+#. Add the needed configuration files in ``config/packages/staging/`` to
+   define the behavior of the new environment. Symfony loads first the files in
+   ``config/packages/*.yaml``, so you must only configure the differences with
+   those files;
```

Comment:
Maybe add a tip that, if the configuration is equal to some other env, using symlinks is a great idea? I.e. in our staging environment, almost everything is production settings, except from `framework.debug` and monolog. So all other `config/packages/prod/*.yaml` are symlinked to `config/packages/staging/*.yaml`